dinov2/distributed/__init__.py

```diff
@@ -90,7 +83,7 @@ def print(*args, **kwargs):
 def _get_master_port(seed: int = 0) -> int:
     MIN_MASTER_PORT, MAX_MASTER_PORT = (20_000, 60_000)
 
-    master_port_str = os.environ.get("MASTER_PORT")
+    master_port_str = os.environ.get("MASTER_PORT") or os.environ.get("SGE_QMASTER_PORT") # For slurm or grid engine
```

Comment:
`SGE_QMASTER_PORT` and `MASTER_PORT` are two different things. I think the former is the port of the qmaster, maybe the process of scheduler, the latter is set by torchrun. Thus I'd exclude the `or os.environ.get("SGE_QMASTER_PORT") # For slurm or grid engine`.